File: AceleraAtos_ProjFinal_KBF/KnowledgeBaseForum.DataAccessLayer/Model/Alerta.cs
namespace KnowledgeBaseForum.DataAccessLayer.Model
{
    public class Alerta : BaseEntity
    {
        /// <summary>
        /// Alert mode (enum).
        /// </summary>
        public int ModoAlerta { get; set; }

        /// <summary>
        /// Author (User) Id.
        /// </summary>
        public string UsuarioId { get; set; } = null!;

        /// <summary>
        /// Whether there is an update or not.
        /// </summary>
        public bool Atualizacao { get; set; }

        /// <summary>
        /// Topic Id.
        /// </summary>
        public Guid TopicoId { get; set; }

        // Relational properties.

        /// <summary>
        /// Topic to which the alert has been registered.
        /// </summary>
        public Topico? Topico { get; set; } = null!;

        /// <summary>
        /// Author user.
        /// </summary>
        public Usuario? Usuario { get; set; } = null!;
    }
}

File: KnowledgeBaseForum.ClientApp/src/app/model/alerta.ts
import { BaseEntity } from "./base-entity";
import { Topico } from "./topico";

export class Alerta extends BaseEntity {
    public modoAlerta!: number;
    public usuarioId!: string;
    public topicoId!: string;
    public topico?: Topico;
    public atualizacao?: boolean;
}
